src/App.js

```diff
@@ -14,11 +14,108 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
-
-export default App;
+export class App extends React.Component {
+  state = {
+    goods: [...goodsFromServer],
+    selectedGoods: [],
+    selectedGoodsNames: [],
+  }
+
+  render() {
+    const { selectedGoods, selectedGoodsNames, goods } = this.state;
+
+    return (
+      <div className="App">
+        <h1>
+          Selected goods:
+          {`\n`}
+          {
+            selectedGoodsNames.length
+              ? selectedGoodsNames.join(', ')
+              : "Anything wasn't select"
+          }
+        </h1>
+
+        <button
+          type="button"
+          onClick={(event) => {
+            this.setState({
+              selectedGoods: [],
+              selectedGoodsNames: [],
+            });
+          }}
+        >
+          Clear
+        </button>
+
+        <ul>
+          {goods.map(good => (
+            <li
+              key={good}
+              className={
+                `${selectedGoodsNames.includes(good) ? 'selected' : ''}`}
+            >
+              {good}
+
+              {' '}
+
+              <button
+                type="button"
+                onClick={(event) => {
+                  const selectedGood = event.target.parentElement;
```

Comment:
don't use event.target.parentElement, just pass the name (good) in your callback